src/main/java/com/aws/greengrass/builtin/services/pubsub/PubSubIPCEventStreamAgent.java

```diff
@@ -107,9 +108,13 @@ public PublishToTopicResponse publish(String topic, byte[] binaryMessage, String
         return handlePublishToTopicRequest(topic, serviceName, Optional.empty(), Optional.of(binaryMessage));
     }
 
+    @SuppressWarnings("PMD.PreserveStackTrace")
     private PublishToTopicResponse handlePublishToTopicRequest(String topic, String serviceName,
                                                                Optional<Map<String, Object>> jsonMessage,
                                                                Optional<byte[]> binaryMessage) {
+        if (topic == null) {
```

Comment:
Do we also want to check if the topic is empty?

Reply:
No, empty is weird, but technically it would work.